### ISSUE TITLE
Keep getting () error

### ISSUE THREAD

**Issue by Juici** [intellij-rust/intellij-rust#2371]
I keep getting [this](https://gist.github.com/Juici/1ed61c63e2f451a3f8d3afbb66dbfaf6) error whenever I try to do code completion.

```
CLion 2018.1 EAP
Build #CL-181.4203.106, built on March 14, 2018
CLion EAP User
Expiration date: April 13, 2018
JRE: 1.8.0_152-release-1136-b20 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.14.25-1-MANJARO
```
```
rustc 1.26.0-nightly (2789b067d 2018-03-06)
cargo 0.26.0-nightly (1d6dfea44 2018-01-26)
rustup 1.11.0 (e751ff9f8 2018-02-13)
```
\
EDIT: Might be related to this? https://github.com/rust-lang-nursery/rls/issues/756
\
EDIT2: Running `rls` command from terminal results
```
$ rls
error: toolchain 'nightly-x86_64-unknown-linux-gnu' does not have the binary `rls`
```

**Comment by Juici** [comment]
Switching toolchain doesn't appear to have helped.
```
rustc 1.24.1 (d3ae9a9e0 2018-02-27)
cargo 0.25.0 (96d8071da 2018-02-26)
```

**Comment by Juici** [comment]
Force reinstalling `rls`
```
$ cargo install rls --force
    Updating registry `https://github.com/rust-lang/crates.io-index`
  Installing rls v0.122.2
   Compiling rls v0.122.2
    Finished release [optimized] target(s) in 0.31 secs
   Replacing ~/.cargo/bin/rls
```
\
`rls` has no output, dunno if it's supposed to?
```
$ rls
$ rls
```
\
Still running into the [same problem](https://gist.github.com/Juici/1ed61c63e2f451a3f8d3afbb66dbfaf6). I am not sure if the `rls` thing is related but you gotta try right?

**Comment by Undin** [comment]
@Juici Thanks for the report!
I've reproduced it too. It's a bug in our parser related code

> EDIT: Might be related to this? rust-lang-nursery/rls#756

No, we don't use `rls` at all so it's our own bug

**Comment by Undin** [comment]
@vlad20012 
It's because of the latest changes in our grammar
Try to type something like
```rust
fn foo(t: () + Qqq) {
}
```

**Comment by Juici** [comment]
For now I've had to switch back to this build: https://plugins.jetbrains.com/plugin/8182-rust/update/43537

**Comment by vlad20012** [comment]
@Undin Ok, I can fix it, but how something like `() + Qqq` can exist in a valid code?

**Comment by vlad20012** [comment]
@Juici could you please specify the rust project in which this bug is reproduced (if it is open-source, of course)?

**Comment by Juici** [comment]
@vlad20012 It wasn't that, it was just regular completion.

I was using the [cursive](https://github.com/gyscos/Cursive) library to try and create a tui application.

eg.
```rust
extern crate cursive;

use cursive::Cursive;

fn main() {
    let mut cur = Cursive::new();

    cur.

    cur.run();
}
```

Trying to code complete on the `cur.` there would result in the error with message `()`.

The code completed fine on [this build](https://plugins.jetbrains.com/plugin/8182-rust/update/43537) but on the newest stable or nightly builds it errors.

**Comment by Undin** [comment]
Fixed in #2375
